pkg/ccl/importccl/import_stmt.go

```diff
@@ -1249,6 +1270,52 @@ func (r *importResumer) ReportResults(ctx context.Context, resultsCh chan<- tree
 	}
 }
 
+// unsupportedStmtConfig stores information that controls how we handle
+// unsupported PGDUMP SQL statements seen during the import.
+type unsupportedStmtConfig struct {
+	// Values are initialized based on the options specified in the IMPORT PGDUMP
+	// stmt.
+	ignoreUnsupported        bool
+	ignoreUnsupportedLogDest string
+
+	// logBuffer holds the string to be flushed to the ignoreUnsupportedLogDest.
+	logBuffer       *bytes.Buffer
+	numIgnoredStmts int
+	// skipLoggingParseErr skips logging stmts that are ignored because we failed
+	// to parse them from the dump file. This is used to avoid logging duplicates
+	// during the schema extraction and data ingestion phases of import.
+	skipLoggingParseErr bool
+}
+
+func logIgnoredStmtsDuringSchemaParsing(
+	ctx context.Context, p sql.JobExecContext, cfg *unsupportedStmtConfig,
+) error {
+	if cfg.ignoreUnsupportedLogDest == "" {
+		return nil
+	}
+
+	numLoggedStmts := pgDumpMaxLoggedStmts
+	if cfg.numIgnoredStmts < pgDumpMaxLoggedStmts {
+		numLoggedStmts = cfg.numIgnoredStmts
+	}
+	cfg.logBuffer.WriteString(fmt.Sprintf("\nLogging %d out of %d ignored stmts.\n",
+		numLoggedStmts, cfg.numIgnoredStmts))
+
+	dest, err := p.ExecCfg().DistSQLSrv.ExternalStorageFromURI(ctx,
+		cfg.ignoreUnsupportedLogDest, p.User())
+	if err != nil {
+		return errors.New("failed to log unsupported stmts during IMPORT PGDUMP")
+	}
+	defer dest.Close()
+	err = dest.WriteFile(ctx, pgDumpUnsupportedSchemaStmtLog,
+		bytes.NewReader(cfg.logBuffer.Bytes()))
+	if err != nil {
+		return errors.New("failed to log unsupported during IMPORT PGDUMP")
```

Comment:
Should we wrap the error here, and above?

Reply:
Good point, done.